curriculum/challenges/english/03-front-end-libraries/react/create-a-controlled-form.md

```diff
@@ -98,6 +98,25 @@ Submitting the form should run `handleSubmit` which should set the `submit` prop
 })();
 ```
 
+`handleSubmit` should call `event.preventDefault`
+
+```js
+const handleSubmit = MyForm.prototype.handleSubmit.toString();
+const allMatches = handleSubmit.match(/\bevent\.preventDefault\(\s*?\)/g) ?? [];
+const blockCommented = handleSubmit.match(
+  /\/\*.*?\bevent\.preventDefault\(\s*?\).*?\*\//gs
+);
+const lineCommented = handleSubmit.match(
+  /\/\/.*?\bevent\.preventDefault\(\s*?\)/g
+);
+const commentedMatches = [...(blockCommented ?? []), ...(lineCommented ?? [])];
+
+assert(
+  // At least one event.preventDefault() call exists and is not commented out
+  allMatches.length && (allMatches.length > commentedMatches.length)
```

Comment:
```suggestion
  allMatches.length > commentedMatches.length
```
We know that `allMatches` has `.length` because it's either a array of matches or `[]`